tests/devices/unit_tests/test_bart_robot.py

```diff
@@ -121,33 +137,33 @@ async def test_given_program_not_running_and_pin_unmounts_then_mounts_when_load_
 
 async def test_given_waiting_for_pin_to_mount_when_no_pin_mounted_then_error_raised():
     device = await _get_bart_robot()
-    status = create_task(device.pin_mounted_or_no_pin_found())
-    await sleep(0.2)
     set_mock_value(device.prog_error.code, 25)
-    await sleep(0.01)
+    status = create_task(device.pin_mounted_or_no_pin_found())
```

Comment:
Could: If we're just relying on the `prog_error` already being set then we needn't spawn this in a new task.